node/node.go

```diff
@@ -91,31 +99,67 @@ func NewNodeController(p NodeProvider, node *corev1.Node, nodes v1.NodeInterface
 	}
 
 	n.nodePingController = newNodePingController(n.p, n.pingInterval, n.pingTimeout)
+	if n.leaseControllerProvider != nil {
+		n.leaseController = n.leaseControllerProvider(n.pingInterval)
+	}
 
 	return n, nil
 }
 
 // NodeControllerOpt are the functional options used for configuring a node
 type NodeControllerOpt func(*NodeController) error // nolint:golint
 
-// WithNodeEnableLeaseV1Beta1 enables support for v1beta1 leases.
-// If client is nil, leases will not be enabled.
-// If baseLease is nil, a default base lease will be used.
-//
-// The lease will be updated after each successful node ping. To change the
-// lease update interval, you must set the node ping interval.
-// See WithNodePingInterval().
-//
-// This also affects the frequency of node status updates:
-//   - When leases are *not* enabled (or are disabled due to no support on the cluster)
-//     the node status is updated at every ping interval.
-//   - When node leases are enabled, node status updates are controlled by the
-//     node status update interval option.
-// To set a custom node status update interval, see WithNodeStatusUpdateInterval().
-func WithNodeEnableLeaseV1Beta1(client v1beta1.LeaseInterface, baseLease *coord.Lease) NodeControllerOpt {
+// WithNodeEnableLeaseV1 enables support for v1 leases.
+// V1 Leases share all the same properties as v1beta1 leases, except they do not fallback like
+// the v1beta1 lease controller does if the API server does not support it.
+func WithNodeEnableLeaseV1(client coordclientset.LeaseInterface, leaseDurationSeconds int32) NodeControllerOpt {
+	if client == nil {
+		panic("client is nil")
+	}
+
 	return func(n *NodeController) error {
-		n.leases = client
-		n.lease = baseLease
+		if n.leaseControllerProvider != nil {
+			return ErrConflictingLeaseControllerConfiguration
+		}
+
+		if leaseDurationSeconds == 0 {
+			leaseDurationSeconds = 40
```

Comment:
If this was a const as suggested up in this review, we could simply reference it here.